substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/events/ThreadCPULoadEvent.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2023, 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, 2023, BELLSOFT. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.jfr.events;
+
+import org.graalvm.nativeimage.StackValue;
+import org.graalvm.nativeimage.IsolateThread;
+
+import com.oracle.svm.core.Uninterruptible;
+import com.oracle.svm.core.jfr.JfrEvent;
+import com.oracle.svm.core.jfr.JfrNativeEventWriter;
+import com.oracle.svm.core.jfr.JfrNativeEventWriterData;
+import com.oracle.svm.core.jfr.JfrNativeEventWriterDataAccess;
+import com.oracle.svm.core.jfr.JfrTicks;
+import com.oracle.svm.core.jfr.JfrThreadLocal;
+
+import com.oracle.svm.core.jdk.Jvm;
+
+import com.oracle.svm.core.thread.VMThreads;
+import com.oracle.svm.core.thread.PlatformThreads;
+import com.oracle.svm.core.thread.ThreadCpuTimeSupport;
+
+import static com.oracle.svm.core.thread.PlatformThreads.IsolateThreadConsumer;
+
+public class ThreadCPULoadEvent {
+
+    private static final int NANOSECS_PER_MILLISEC = 1_000_000;
+    private static volatile int lastActiveProcessorCount;
+
+    private static final IsolateThreadConsumer threadCPULoadEventConsumer = new IsolateThreadConsumer() {
+
+        @Uninterruptible(reason = "Thread locks/holds the THREAD_MUTEX.", mayBeInlined = true)
```

Comment:
I think in this case it doesn't matter because only `PlatformThreads.iterateIsolateThreads` calls this, but generally, if code is meant to be holding a `VMMutex` it should ensure the caller is also uninterruptible to guarantee the whole critical section is uninterruptible. And should additionally avoid `mayBeInlined = true` because if it is inlined into a interruptible code, it will unexpectedly become interruptible as well. There's a good [summary here](https://github.com/oracle/graal/pull/5384#discussion_r1094805321) of uninterruptible use cases.